core/new-gui/src/app/workspace/component/visualization-panel-content/visualization-panel-content.component.ts

```diff
@@ -102,6 +119,95 @@ export class VisualizationPanelContentComponent implements AfterViewInit, OnDest
       case ChartType.SPLINE:
         this.generateChart();
         break;
+      case ChartType.SPATIAL_SCATTERPLOT:
+        this.generateSpatialScatterplot();
+        break;
+      case ChartType.SCATTERPLOT:
+        this.generateSimpleScatterplot();
+        break;
+    }
+  }
+
+  generateSimpleScatterplot() {
+    if (this.c3ChartElement) {
+      this.c3ChartElement.destroy();
+    }
+    const result = this.data as Array<Record<string, Primitive>>;
+    const xLabel: string = Object.keys(result[0])[0];
+    const yLabel: string = Object.keys(result[0])[1];
+
+    this.c3ChartElement = c3.generate({
+      size: {
+        height: VisualizationPanelContentComponent.HEIGHT,
+        width: VisualizationPanelContentComponent.WIDTH
+      },
+      data: {
+        json: result,
+        keys: {
+          x: xLabel,
+          value: [yLabel]
+        },
+        type: this.chartType as c3.ChartType
+      },
+      axis: {
+        x: {
+          label: xLabel,
+          tick: {
+            fit: true
+          }
+        },
+        y: {
+          label: yLabel
+        }
+      },
+      bindto: VisualizationPanelContentComponent.CHART_ID
+    });
+  }
+
+  generateSpatialScatterplot() {
+    if (this.map === undefined) {
+      this.initMap();
+    }
+    this.map?.on('styledata', () => {
+      this.addScatterLayer();
+    });
+  }
+
+  initMap() {
+    /* mapbox object with default configuration */
+    this.map = new mapboxgl.Map({
+      container: VisualizationPanelContentComponent.MAP_CONTAINER,
+      style: 'mapbox://styles/mapbox/light-v9',
+      center: [-96.35, 39.5],
+      zoom: 3,
+      maxZoom: 17,
+      minZoom: 0
+    });
+  }
+
+  addScatterLayer() {
+    if (this.map !== undefined) {
+      if (this.map?.getLayer('scatter')) {
+        this.map?.removeLayer('scatter');
+      }
```

Comment:
if we are removing the previous `scatter` layer, then maybe the method name can be updated? by the method name, I assumed the new layer is being appended not replacing the old layer.

Reply:
in this PR I am always replacing the old layer by deleting it then adding a new layer once PR #1100 is merged, I will append new layers not replace

Reply:
Currently your method name does not match the method behavior, it will be good to update one of them.

Reply:
what is the method name that doesn't match the behavior? pleas specify? `removeLayer` is to remove a layer `getLayer` is to make sure the layer exists otherwise the library will fire an exception if you try to remove a layer that doesn't exist. These functions are the library's I am just calling them. `scatter` is the name of the layer

Reply:
The add_layer method